File: eRecipes/eRecipes.Model/Korisnik.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace eRecipes.Model
{
    public partial class Korisnik
    {
        public int KorisnikId { get; set; }

        public string Ime { get; set; } = null!;

        public string Prezime { get; set; } = null!;

        public DateTime? DatumRodjenja { get; set; }

        public string? Email { get; set; }

        public string? Telefon { get; set; }

        public string KorisnickoIme { get; set; } = null!;
        //public string Uloge => string.Join(", ", KorisnikUlogas?.Select(x => x.Uloga?.Naziv)?.ToList());
        public int? UlogaId { get; set; }
        public Uloga Uloga { get; set; }

        public bool? Status { get; set; }
    }
}

File: eRecipes/UI/erecipes_desktop/lib/models/korisnik.dart
import 'package:erecipes_desktop/models/uloga.dart';
import 'package:json_annotation/json_annotation.dart';

part 'korisnik.g.dart';

@JsonSerializable()
class Korisnik{
  int? korisnikId;
  String? ime; 
  String? prezime; 
  DateTime? datumRodjenja;
   String? email; 
  String? telefon; 
  String? korisnickoIme; 
  String? lozinka; 
  Uloga? uloga; 
  String? uloge;
//List<KorisnikUloga> korisnikUlogas;
   Korisnik({
    this.ime,
    this.prezime,
    this.datumRodjenja,
    this.email,
    this.telefon,
    this.korisnickoIme,
    this.lozinka,
    this.uloga,
    this.uloge
   });

  factory Korisnik.FromJson(Map<String,dynamic> json)=> _$KorisnikFromJson(json);

  Map<String,dynamic> toJson() => _$KorisnikToJson(this);
}